Carbon.Cryptography/Functions/ConvertTo-ProviderAccessControlRights.ps1

```diff
@@ -0,0 +1,59 @@
+
+function ConvertTo-ProviderAccessControlRights
+{
+    <#
+    .SYNOPSIS
+    Converts strings into the appropriate access control rights for a PowerShell provider (e.g. FileSystemRights or
+    RegistryRights).
+
+    .DESCRIPTION
+    This is an internal Carbon function, so you're not getting anything more than the synopsis.
+
+    .EXAMPLE
+    ConvertTo-ProviderAccessControlRights -ProviderName 'FileSystem' -InputObject 'Read','Write'
+
+    Demonstrates how to convert `Read` and `Write` into a `System.Security.AccessControl.FileSystemRights` value.
+    #>
+    [CmdletBinding()]
+    param(
+        # The provider name.
+        [Parameter(Mandatory)]
+        [ValidateSet('FileSystem', 'CryptoKey')]
+        [String] $ProviderName,
+
+        # The values to convert.
+        [Parameter(Mandatory, ValueFromPipeline)]
+        [ValidateSet('Read', 'FullControl')]
+        [String] $InputObject
+    )
+
+    process
+    {
+        Set-StrictMode -Version 'Latest'
+        Use-CallerPreference -Cmdlet $PSCmdlet -Session $ExecutionContext.SessionState
+
+        if ($providerName -eq 'CryptoKey')
+        {
+            # CryptoKeyRights
+            # Read          0x80100000  GenericRead, Synchronize
+            # FullControl   0x90100000  GenericRead, GenericAll, Synchronize
+            $rights = [Security.AccessControl.CryptoKeyRights]::GenericRead
+            if ($Permission -eq 'FullControl')

```

Comment:
Turns out, this was only working because `$Permission` was assigned a value in both the parent scopes that `ConvertTo-ProviderAccessControlRights` was being called from. In researching, I've discovered this is only called to convert permissions to CryptoKeyRights, so I'm renaming this function to `ConvertTo-CryptoKeyRights`.